framework/include/mediaplayer/mediaplayer.h

```diff
@@ -0,0 +1,224 @@
+#ifndef __MEDIAPLAYER_H__
+#define __MEDIAPLAYER_H__
+
+#include <queue.h>
+#include <sys/types.h>
+
+
+
+/***
+ * Enumerations
+ */
+
+typedef enum {
+	MEDIA_OK,		
+	MEDIA_ERROR,
+	MEDIA_ERROR_THREAD_INIT,
+	MEDIA_ERROR_PCM_WRITE,
+	MEDIA_ERROR_WRITE_TO_FILE,
+	MEDIA_ERROR_CREATE,
+	MEDIA_ERROR_UNKNOWN_FILE
+} media_result_e;
+
+
+
+typedef enum {
+	MEDIA_TYPE_PCM,
+	MEDIA_TYPE_WAV
+} media_type_e;
+
+
+typedef enum {
+	MEDIA_PLAY_TYPE_FILE,
+	MEDIA_PLAY_TYPE_NETWORK,
+} media_play_type_e;
+
+typedef enum {
+	MEDIA_OP_RECORD,
+	MEDIA_OP_PLAYBACK
+} media_op_e;
+
+typedef enum {
+	MEDIA_STATE_CREATED,
+	MEDIA_STATE_PLAYING,
+	MEDIA_STATE_PAUSING,
+	MEDIA_STATE_PAUSED,
+	MEDIA_STATE_STOPPING,
+	MEDIA_STATE_RECORDING,
+	MEDIA_STATE_CLOSING
+} media_state_e;
+
+/***
+ * Structures
+ */
+struct media_s;
+typedef struct media_s media_t;
+
+struct media_cb_s {
+	void (*on_finished)(media_t *m);
+};
+
+/****************************************************************************
+ * Public Function Prototypes
+ ****************************************************************************/
+
+#ifdef __cplusplus
+#define EXTERN extern "C"
+extern "C" {
+#else
+#define EXTERN extern
+#endif
+
+/**
+ * @brief media_init() Initialize mediaplayer.
```

Comment:
You can add these doxygen comments when you prepare tc for this. Before making it, let's add cond, internal.